src/ChannelServer/Network/ChannelPacketSender.cs

```diff
@@ -4105,14 +4105,105 @@ public static void ZC_UPDATED_MONSTERAPPEARANCE(Monster monster)
 			monster.Map.Broadcast(packet, monster);
 		}
 
+		/// <summary>
+		/// Display a buff on client UI
+		/// </summary>
+		/// <param name="entity"></param>
+		/// <param name="buff"></param>
+		public static void ZC_BUFF_ADD(IEntity entity, Buff buff)
+		{
+			var packet = new Packet(Op.ZC_BUFF_ADD);
+			packet.AddBuff(buff);
+
+			entity.Map.Broadcast(packet);
+		}
+
+		/// <summary>
+		/// Update a buff after a buff has been added
+		/// </summary>
+		/// <param name="entity"></param>
+		/// <param name="buff"></param>
+		public static void ZC_BUFF_UPDATE(IEntity entity, Buff buff)
+		{
+			var packet = new Packet(Op.ZC_BUFF_UPDATE);
+			packet.AddBuff(buff);
+
+			entity.Map.Broadcast(packet);
+		}
+
+		/// <summary>
+		/// Remove buff from client UI
+		/// </summary>
+		/// <param name="entity"></param>
+		/// <param name="buff"></param>
+		public static void ZC_BUFF_REMOVE(IEntity entity, Buff buff)
+		{
+			var packet = new Packet(Op.ZC_BUFF_REMOVE);
+			packet.PutInt(entity.Handle);
+			packet.PutInt((int)buff.Id);
+			packet.PutByte(0);
+			packet.PutInt(buff.Handle);
+			packet.PutByte(0);
+
+			entity.Map.Broadcast(packet);
+		}
+
+		/// <summary>
+		/// Buff list, sent when a entity spawns?
+		/// </summary>
+		/// <param name="entity"></param>
+		public static void ZC_BUFF_LIST(IEntity entity)
+		{
+			var packet = new Packet(Op.ZC_BUFF_REMOVE);
```

Comment:
Is `ZC_BUFF_REMOVE` right? The method is named `ZC_BUFF_LIST`. Same for `ZC_BUFF_CLEAR`.

Reply:
I've fixed this implementation. ZC_BUFF_LIST is usually sent on entity entering the map.
ZC_BUFF_CLEAR is sent on entity death usually, maybe when it leaves the map too?